Packages/com.chisel.components/Chisel/Components/API.public/BrushMeshAssetFactory/Cylinder/BrushFactory.Cylinder.cs

```diff
@@ -19,18 +19,40 @@ namespace Chisel.Components
     // TODO: rename
     public sealed partial class BrushMeshAssetFactory
     {
-        public static bool GenerateCylinderAsset(CSGBrushMeshAsset brushMeshAsset, CSGCylinderDefinition bottom, float topHeight, float rotation, int sides, CSGSurfaceAsset[] surfaceAssets, SurfaceDescription[] surfaceDescriptions)
+        public static bool GenerateCylinderAsset(CSGBrushMeshAsset brushMeshAsset, CSGCylinderDefinition definition)
         {
-            CSGCylinderDefinition top;
+            definition.Validate();
+
+            var tempTop		= definition.top;
+            var tempBottom	= definition.bottom;
+
+            if (!definition.isEllipsoid)
+            {
+                tempTop.diameterZ = tempTop.diameterX;
+                tempBottom.diameterZ = tempBottom.diameterX;
+            }
+
+            switch (definition.type)
+            {
+                case CylinderShapeType.Cylinder:		return BrushMeshAssetFactory.GenerateCylinderAsset(brushMeshAsset, tempBottom, tempTop.height, definition.rotation, definition.sides, definition.surfaceAssets, definition.surfaceDescriptions); 
+                case CylinderShapeType.ConicalFrustum:	return BrushMeshAssetFactory.GenerateConicalFrustumAsset(brushMeshAsset, tempBottom, tempTop, definition.rotation, definition.sides, definition.surfaceAssets, definition.surfaceDescriptions); 
+                case CylinderShapeType.Cone:			return BrushMeshAssetFactory.GenerateConeAsset(brushMeshAsset, tempBottom, tempTop.height, definition.rotation, definition.sides, definition.surfaceAssets, definition.surfaceDescriptions); 
+            }
+            return false;
+        }
+
+        public static bool GenerateCylinderAsset(CSGBrushMeshAsset brushMeshAsset, CSGCircleDefinition bottom, float topHeight, float rotation, int sides, CSGSurfaceAsset[] surfaceAssets, SurfaceDescription[] surfaceDescriptions)
+        {
+            CSGCircleDefinition top;
             top.diameterX = bottom.diameterX;
             top.diameterZ = bottom.diameterZ;
             top.height = topHeight;
             return GenerateConicalFrustumAsset(brushMeshAsset, bottom, top, rotation, sides, surfaceAssets, surfaceDescriptions);
         }
 
-        public static bool GenerateConeAsset(CSGBrushMeshAsset brushMeshAsset, CSGCylinderDefinition bottom, float topHeight, float rotation, int sides, CSGSurfaceAsset[] surfaceAssets, SurfaceDescription[] surfaceDescriptions)
+        public static bool GenerateConeAsset(CSGBrushMeshAsset brushMeshAsset, CSGCircleDefinition bottom, float topHeight, float rotation, int sides, CSGSurfaceAsset[] surfaceAssets, SurfaceDescription[] surfaceDescriptions)
         {
-            CSGCylinderDefinition top;
+            CSGCircleDefinition top;
```

Comment:
Renamed CSGCylinderDefinition to CSGCircleDefinition, which was only a 2D circle/ellipsoid definition anyway. This way I could use CSGCylinderDefinition to define a .. cylinder